src/db/transaction.cr

```diff
@@ -0,0 +1,118 @@
+module DB
+  abstract class Transaction
+    include Disposable
+    include BeginTransaction
+
+    abstract def connection : Connection
+
+    def commit
+      close!
+    end
+
+    def rollback
+      close!
+    end
+
+    private def close!
+      raise DB::Error.new("Transaction already closed") if closed?
```

Comment:
Well, I prefer caution here. If a TopTransaction ref somehow leak at hold a reference to an open connection it could commit/rollback a not intended transaction. It should not happen, unless bugs, but let's keep this.